### ISSUE TITLE
Slow training when adding more workers

### ISSUE THREAD

**Issue by Devanish31** [MIC-DKFZ/nnUNet#2965]
Starting training run with the standard nnUnet configuration with few changes: 
    "fg_oversample": 0.9,
    "train_iter": 250,
    "val_iter": 50

2025-12-22 03:13:45.616015: Unable to plot network architecture: nnUNet_compile is enabled!
2025-12-22 03:13:45.702593:
2025-12-22 03:13:45.704159: Epoch 0
2025-12-22 03:13:45.705711: Current learning rate: 0.01
2025-12-22 03:16:06.958722: Mean training foreground fraction this epoch: 0.029615%
2025-12-22 03:16:25.743766: train_loss -0.1749
2025-12-22 03:16:25.745121: val_loss -0.3077
2025-12-22 03:16:25.746762: Pseudo dice [np.float32(0.5172)]
2025-12-22 03:16:25.748258: Epoch time: 160.04 s
2025-12-22 03:16:26.138250: Yayy! New best EMA pseudo Dice: 0.5171999931335449
2025-12-22 03:16:27.614910:
2025-12-22 03:16:27.616590: Epoch 1
2025-12-22 03:16:27.618119: Current learning rate: 0.00999
2025-12-22 03:17:56.305979: Mean training foreground fraction this epoch: 0.028157%
2025-12-22 03:18:00.796844: train_loss -0.4513
2025-12-22 03:18:00.798167: val_loss -0.4922
2025-12-22 03:18:00.799669: Pseudo dice [np.float32(0.7577)]
2025-12-22 03:18:00.801163: Epoch time: 93.19 s
2025-12-22 03:18:01.696419: Yayy! New best EMA pseudo Dice: 0.5412999987602234
2025-12-22 03:18:03.166559:
2025-12-22 03:18:03.168220: Epoch 2
2025-12-22 03:18:03.169715: Current learning rate: 0.00999

Then added more workers:
export nnUNetv2_trainer_dataloader_num_workers=22

2025-12-22 03:18:36.512992: Unable to plot network architecture: nnUNet_compile is enabled!
2025-12-22 03:18:36.527525:
2025-12-22 03:18:36.529217: Epoch 0
2025-12-22 03:18:36.530747: Current learning rate: 0.01
2025-12-22 03:20:16.845983: Mean training foreground fraction this epoch: 0.031373%
2025-12-22 03:20:25.267099: train_loss -0.0081
2025-12-22 03:20:25.268371: val_loss -0.1632
2025-12-22 03:20:25.270001: Pseudo dice [np.float32(0.0859)]
2025-12-22 03:20:25.271430: Epoch time: 108.74 s
2025-12-22 03:20:25.629299: Yayy! New best EMA pseudo Dice: 0.08590000122785568
2025-12-22 03:20:26.932404:
2025-12-22 03:20:26.934184: Epoch 1
2025-12-22 03:20:26.935636: Current learning rate: 0.00999
2025-12-22 03:21:55.571339: Mean training foreground fraction this epoch: 0.028027%
2025-12-22 03:22:00.053386: train_loss -0.4487
2025-12-22 03:22:00.054729: val_loss -0.6434
2025-12-22 03:22:00.056343: Pseudo dice [np.float32(0.6748)]
2025-12-22 03:22:00.058080: Epoch time: 93.12 s
2025-12-22 03:22:00.789674: Yayy! New best EMA pseudo Dice: 0.14480000734329224
2025-12-22 03:22:02.207714:
2025-12-22 03:22:02.209413: Epoch 2
2025-12-22 03:22:02.211000: Current learning rate: 0.00999
2025-12-22 03:23:31.252797: Mean training foreground fraction this epoch: 0.031037%
2025-12-22 03:23:35.769226: train_loss -0.6197
2025-12-22 03:23:35.770681: val_loss -0.6222
2025-12-22 03:23:35.772309: Pseudo dice [np.float32(0.6774)]
2025-12-22 03:23:35.773951: Epoch time: 93.56 s
2025-12-22 03:23:37.890655: Yayy! New best EMA pseudo Dice: 0.1979999989271164
2025-12-22 03:23:39.303439:
2025-12-22 03:23:39.305158: Epoch 3
2025-12-22 03:23:39.306682: Current learning rate: 0.00998
2025-12-22 03:25:08.503506: Mean training foreground fraction this epoch: 0.029973%
2025-12-22 03:25:13.036195: train_loss -0.6282
2025-12-22 03:25:13.037386: val_loss -0.6437
2025-12-22 03:25:13.038923: Pseudo dice [np.float32(0.7161)]
2025-12-22 03:25:13.040505: Epoch time: 93.73 s
2025-12-22 03:25:13.429651: Yayy! New best EMA pseudo Dice: 0.24979999661445618
2025-12-22 03:25:14.859528:
2025-12-22 03:25:14.861181: Epoch 4
2025-12-22 03:25:14.862580: Current learning rate: 0.00998
2025-12-22 03:26:43.964372: Mean training foreground fraction this epoch: 0.032898%
2025-12-22 03:26:48.483428: train_loss -0.6564
2025-12-22 03:26:48.484653: val_loss -0.717
2025-12-22 03:26:48.486213: Pseudo dice [np.float32(0.7656)]
2025-12-22 03:26:48.487766: Epoch time: 93.63 s
2025-12-22 03:26:48.877409: Yayy! New best EMA pseudo Dice: 0.30140000581741333
2025-12-22 03:26:50.315412:
2025-12-22 03:26:50.317060: Epoch 5
2025-12-22 03:26:50.318562: Current learning rate: 0.00997

What is the reason for the huge difference between the starting best EMA pseudo Dice after adding workers?
Is there a reason or simply random event?
